zkbind/src/org/zkoss/bind/xel/zel/ListModelELResolver.java

```diff
@@ -41,12 +44,29 @@ public Object getValue(ELContext context, Object base, Object property)
 		if (base instanceof ListModel<?>) {
 			ListModel<?> listmodel = (ListModel<?>) base;
 			Integer idx = coerce(property);
+			boolean itemIdx = true;
 			if (idx == null) { // property is not a legal number format
-				return null; // unresolved null
+				itemIdx = false;
+				//ZK-3749 maybe it is a BeanEL
+				//BeanELResolver beanELResolver = new BeanELResolver();
+				//Object result = beanELResolver.getValue(context, base, property);
+				if (property instanceof String) {
+					try {
+						property = Fields.get(base, (String) property);
+					} catch (NoSuchMethodException e) {
+
+					}
+					idx = coerce(property);
+				}
+				if (idx == null) {
+					return null;
+				}

```

Comment:
A workaround in `ListModelELResolver` is absolutely wrong. In fact, `ListELResolver` is wrong (https://issues.apache.org/jira/browse/EL-3). If the property is not a number, it should just return null (and not throw an exception) and let `BeanELResolver` handle it.